library/modules/bigip_provision.py

```diff
@@ -51,21 +54,37 @@
         For example, changing one module to C(dedicated) requires setting all
         others to C(none). Setting the level of a module to C(none) means that
         the module is not activated.
-      - This parameter is not relevant to C(cgnat) and will not be applied to the
-        C(cgnat) module.
+      - Use c(state) absent to set c(level) to none and de-provision module.
+      - This parameter is not relevant to C(cgnat - pre tmos 15.0) or C(mgmt) and will not be
+        applied to the C(cgnat - pre tmos 15.0) or C(mgmt) module.
     type: str
     choices:
       - dedicated
       - nominal
       - minimum
     default: nominal
+  memory:
+    description:
+      - Sets additional memory for Management Module.
+      - Options include small, medium, large, or integer between 0 and maximum available ram.
+      - Use Large for configurations containing more than 2000 objects, or
+        more specifically, for any configuration that exceeds 1000 objects
+        per 2 GB of installed memory. Changing the Management C(mgmt) size
+        after initial provisioning causes a reprovision operation
+    type: str
+    choices:
+      - small
+      - medium
+      - large
+      - integer between 0 and maximum available ram.
```

Comment:
you are missing version added, you needs something like this, but in our case the version is 2.9:

https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_device_traffic_group.py#L72